craft_application/_config.py

```diff
@@ -0,0 +1,37 @@
+# This file is part of craft-application.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Configuration model for craft applications."""
+from __future__ import annotations
+
+import craft_cli
+import pydantic
+
+
+class ConfigModel(pydantic.BaseModel):
```

Comment:
Shouldn't this file go under `craft_application/models/`?

Reply:
I didn't do this because it's not a model for use with YAML files or related, but rather something internal.